src/isomorphic/modern/class/ReactBaseClasses.js

```diff
@@ -144,7 +144,29 @@ ReactPureComponent.prototype.constructor = ReactPureComponent;
 Object.assign(ReactPureComponent.prototype, ReactComponent.prototype);
 ReactPureComponent.prototype.isPureReactComponent = true;
 
+function ReactAsyncComponent(props, context, updater) {
+  // Duplicated from ReactComponent.
+  this.props = props;
+  this.context = context;
+  this.refs = emptyObject;
+  // We initialize the default updater but the real one gets injected by the
+  // renderer.
+  this.updater = updater || ReactNoopUpdateQueue;
+}
+
+function ComponentDummy2() {}
```

Comment:
Can we reuse the dummy function from above? Not sure if that is worth from an optimization perspective or not.